Incorporated vidcdn updates

JRCdev opened the pull request:
Improvements added:
 - Added ajax code matching current linux master
 - updated download code to use aria2c if user's got it on their path but if they don't it'll use curl
 - Filters out .vtt files that sometimes show up in the list of download links

Limitations:
 From my limited testing it still gets the odd 403 errors with some series and settings but not others:
 - Fine for the FLCL dubs but not subs
 - Fine for Serial Experiments Lain but not Legend of the Galactic Heroes
But with this it once again plays _many_ things as opposed to _no_ things. If these bugs also exist in linux version this'll get windows users back to that level of access.

CoolnsX commented on ani-cli-win:
```diff
@@ -213,7 +229,9 @@ open_episode () {
 	video_url=$(get_links "$dpage_link")
+	# printf "%s\n" "$video_url"
 
 	case $video_url in
```
Can u remove from line 236 to 247.. these lines are useless.. it doesn't do anything..

CoolnsX replied:
Also from line 92 to 95.. cant't we use jq??..
Because these line will be helpful in new anime.. but in older anime.. it will produce useless link..
I know adding jq as dependency will be a problem in windows.. but if the script is not up to the task then it will make the windows user ditch up the idea of ani-cli

JRCdev replied:
Oh, sure!
Also, I could make it so it only uses jq if the user has jq installed, would that be an alright compromise? I'm not sure what the best UX decision is.

ura43 replied:
No. Definitely don't support both. If jq is needed elsewhere in the windows branch already, you might as well port the parsing to jq, but if windows got spared with that dependency then don't change it.
Installing software on windows is more of a hassle and it is safe to assume our average windows user to be less technically literate than our average linux user.
Removing the comments does make sense tho

ura43 commented on ani-cli-win:
```diff
@@ -72,32 +72,48 @@ get_dpage_link() {
+				err "Current video quality is not available (defaulting to best quality)"
 				quality=best
-				video_link=$(printf '%s' "$video_links" | tail -n 1)
+				video_link=$(printf '%s' "$video_links" | head -n 4 | tail -n 1)
```
Why does this need changing? Is the video link now always in line 4, with padding below sometimes?

ura43 replied:
Oh, this is just backporting fixes to the windows branch. My bad

ura43 commented on ani-cli-win:
```diff
@@ -234,15 +252,16 @@ open_episode () {
 		" "$logfile" > "${logfile}.new" && mv "${logfile}.new" "$logfile"
 
-		$player_fn "$video_url" "vlc://quit" --http-referrer="$dpage_link"  --http-user-agent="Mozilla/5.0" --adaptive-use-access  >/dev/null 2>&1
+		nohup "$player_fn" --http-referrer="$dpage_link" --http-user-agent="Mozilla/5.0" --adaptive-use-access "$video_url" "vlc://quit" > /dev/null 2>&1 &
```
This is part of #256 but could you remove the nohup ampersands while you're at it?
They are confirmed not necessary on all supported platforms

ura43 replied:
Never mind, since this is a backport...

ura43 commented on ani-cli-win:
```diff
@@ -234,15 +252,16 @@ open_episode () {
 		{
-			curl -4 -L -# -C - "$video_url" -G -e 'https://streamani.io/' \
-				-o "${anime_id}-${episode}.mp4" "$video_url" >/dev/null 2>&1 &&
+			(command -v aria2c &&
+			aria2c -x 16 -s 16 --referer="$dpage_link" "$video_url" --dir="$download_dir" -o "${anime_id}-${episode}.mp4" --download-result=hide ||
+			curl -4 -L -o "${anime_id}-${episode}.mp4" "$video_url" >/dev/null 2>&1 ) &&
```
I love this syntax. 
`test && optional feature || default behaviour` should be more common in our codebase

ura43 reviewed (changes requested):
Good work so far

ura43 commented:
I didn't spot this was backporting fixes to the windows branch. Please ignore my requested changes from above

JRCdev commented:
I removed the comments and added jq. The latter didn't add or remove functionality afaik, but I've discovered that more of my test cases work when asking for the worst quality explicitly.
